src/mixins/filing-template-mixin.ts

```diff
@@ -301,6 +302,69 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
     return filing
   }
 
+  /**
+   * Builds an special resolution filing from store data.
+   * @param isDraft whether this is a draft
+   * @returns the resolution filing body
+   */
+  buildSpecialResolutionFiling (isDraft: boolean): SpecialResolutionFilingIF {
+    // if filing...
+    if (!isDraft) {
+      // to add in future
+    }
+
+    // Build alteration filing
+    const filing: SpecialResolutionFilingIF = {
+      header: {
+        name: FilingTypes.SPECIAL_RESOLUTION,
+        certifiedBy: this.getCertifyState.certifiedBy,
+        date: this.getCurrentDate, // "absolute day" (YYYY-MM-DD in Pacific time)
+        folioNumber: this.getFolioNumber
+      },
+      business: {
+        foundingDate: this.getEntitySnapshot.businessInfo.foundingDate,
+        identifier: this.getEntitySnapshot.businessInfo.identifier,
+        legalName: this.getEntitySnapshot.businessInfo.legalName,
+        legalType: this.getEntitySnapshot.businessInfo.legalType,
+        nrNumber: this.getEntitySnapshot.businessInfo.nrNumber
+      },
+      alteration: {
+        business: {
+          identifier: this.getBusinessId,
+          legalType: this.getEntityType
+        },
+        provisionsRemoved: this.areProvisionsRemoved,
+        contactPoint: this.getContactPoint
+      }
+    }
+
+    // Apply name request info to filing
+    if (this.getNameRequestNumber || this.hasBusinessNameChanged || this.hasBusinessTypeChanged) {
```

Comment:
Not needed?

Reply:
we need this since we are using a name request component, COOP can change a name using name request, correct me i am wrong

Reply:
Keep it if you need it. I don't know if coops can change their name.

Reply:
yeah, As per design coop can change their name

![Screen Shot 2022-07-26 at 11 31 09 AM](https://user-images.githubusercontent.com/55506249/181084372-9392d835-0c6d-4d98-a771-f52a7d34e958.png)

Reply:
OK, FYI: Lewis has changed some things in the name request mixin that you will need to update. Specifically, you will need to state which NR types are valid for your scenario.

Ref: https://github.com/bcgov/business-edit-ui/pull/333

Reply:
Good to know, I will add reference in name change ticket https://app.zenhub.com/workspaces/relationships-team-space-61435c47e483c4000f08e9f6/issues/bcgov/entity/12894